lua/nvim-tree/api.lua

```diff
@@ -207,9 +207,14 @@ Api.node.navigate.sibling.first = wrap_node(require("nvim-tree.actions.moves.sib
 Api.node.navigate.sibling.last = wrap_node(require("nvim-tree.actions.moves.sibling").fn "last")
 Api.node.navigate.parent = wrap_node(require("nvim-tree.actions.moves.parent").fn(false))
 Api.node.navigate.parent_close = wrap_node(require("nvim-tree.actions.moves.parent").fn(true))
+
+---@class ApiNodeNavigateOpts
+---@field recurse boolean|nil default false
+
 Api.node.navigate.git.next = wrap_node(require("nvim-tree.actions.moves.item").fn("next", "git"))
-Api.node.navigate.git.prev = wrap_node(require("nvim-tree.actions.moves.item").fn("prev", "git"))
 Api.node.navigate.diagnostics.next = wrap_node(require("nvim-tree.actions.moves.item").fn("next", "diag"))
+
+Api.node.navigate.git.prev = wrap_node(require("nvim-tree.actions.moves.item").fn("prev", "git"))
```

Comment:
Ouch... this doesn't quite work as we expect:  when `opts` is not passed the node will be injected.

We've hit the same roadblock as #2583

The solution above was to add specific new API: https://github.com/nvim-tree/nvim-tree.lua/pull/2583/files#diff-b5b2d1abcb86ead5a2319f7e065a7545b834a407cc9f420590d7acebb285fca9R1865

We'll just have to cope with the extra API. Apologies for steering you on the wrong path.

Reply:
Alright, NP. I'll try to review my changes, rebase and update the PR in the coming days/weeks depending on the time I have.

Reply:
Thanks Mate

There is no hurry, this is a tricky one.